spec/features/user/user_adds_song_to_playlist_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+context 'the user is logged in' do
+  context 'the user made a playlist' do
+    scenario 'they can add from a song show page' do
+      user = User.create(username: 'TheRealIanDouglas', password: 'supermutt')
+      playlist = user.playlists.create(name: 'Road')
+      album = Album.create(name: 'spacelounge')
+      song = album.songs.create(name: 'all star')
+
+      visit login_path
+      fill_in 'session[username]', with: user.username
+      fill_in 'session[password]', with: user.password
+      click_on 'Login'
```

Comment:
usually for big tests like this I like to do a `context` of "user is logged in" and have a `before_each` block do the actual logging in. Then I can also have a context block of "user is not logged in" where I can also test that an anonymous (not logged in) user cannot perform similar tasks.